test/bigtest/tests/notice-policy/notice-policy-detail-test.js

```diff
@@ -0,0 +1,703 @@
+              getBooleanRepresentation(noticePolicy.loanNotices[0].realTime)
+            );
+          });
+
+          it('should have a proper label', () => {
+            expect(NoticePolicyDetail.loanNoticesSection.loanNotices(0).realTime.label.text).to.equal(
+              translation['settings.noticePolicy.notices.realTime']
+            );
+          });
+        });
+      });
+
+      describe('send every', () => {
+        describe('notice policy\n\t\t-Recurring', () => {
+          let patronNoticeTemplate;
+          let noticePolicy;
+
+          beforeEach(function () {
+            patronNoticeTemplate = this.server.create('templates', { category: 'Loan' });
+            noticePolicy = this.server.create('patronNoticePolicy', {
+              active: true,
+              loanNotices: [{
+                name: 'mockName',
+                templateId: patronNoticeTemplate.id,
+                templateName: 'mockTemplateName',
+                format: 'Email',
+                frequency: 'Recurring',
+                realTime: true,
+                sendOptions: {
+                  sendHow: 'After',
+                  sendWhen: 'Due date',
+                  sendBy: {
+                    duration: 2,
+                    intervalId: 'Hours'
+                  },
+                  sendEvery: {
+                    duration: 2,
+                    intervalId: 'Hours'
+                  }
+                }
+              }],
+            });
+
+            this.visit(`/settings/circulation/notice-policies/${noticePolicy.id}`);
+          });
+
+          describe('duration', () => {
+            it('should be displayed', () => {
+              expect(NoticePolicyDetail.loanNoticesSection.loanNotices(0).sendEveryDuration.isPresent).to.be.true;
+            });
+
+            it('should have a proper value', () => {
+              expect(NoticePolicyDetail.loanNoticesSection.loanNotices(0).sendEveryDuration.text).to.equal(
+                noticePolicy.loanNotices[0].sendOptions.sendEvery.duration.toString()
+              );
+            });
+          });
+
+          describe('interval', () => {
+            it('should be displayed', () => {
+              expect(NoticePolicyDetail.loanNoticesSection.loanNotices(0).sendEveryIntervalId.isPresent).to.be.true;
+            });
+
+            it('should have a proper value', () => {
+              expect(NoticePolicyDetail.loanNoticesSection.loanNotices(0).sendEveryIntervalId.text).to.equal(
+                translation[
+                  `settings.noticePolicy.notices.${noticePolicy.loanNotices[0].sendOptions.sendEvery.intervalId.toLowerCase()}`
+                ]
+              );
+            });
+          });
+        });
+
+        describe('notice policy\n\t\t-One time', () => {
+          let patronNoticeTemplate;
+          let noticePolicy;
+
+          beforeEach(function () {
+            patronNoticeTemplate = this.server.create('templates', { category: 'Loan' });
```

Comment:
I see that content here is pretty repeating across file with minor changes to the `loanNotices` field. I am thinking about creating a function which can have reasonable defaults to create a basic setup with the ability to extend it where needed via params and call it in the `beforeEach`.

Reply:
I think it's unnecessary, cause this kind creation notice policy in each block brings more clarity to the tests. Because in this case, we rewriting default values from the factory.